### ISSUE TITLE
keep_matching_keys functionality

### ISSUE THREAD

**Issue by jmcdade11** [open-telemetry/opentelemetry-collector-contrib#32989]
### Component(s)

processor/transform

### Is your feature request related to a problem? Please describe.

In the process of moving span event attributes onto the span itself, I only wanted a subset of attributes matching on a prefix.  In my particular case, I have span events for C# mongodb driver events: https://mongodb.github.io/mongo-csharp-driver/2.1/apidocs/html/N_MongoDB_Driver_Core_Events.htm

I knew the prefix was always "db.mongodb" but keep_keys doesn't operate that way.  I ended up for the time being just pulling in the entire set of attributes.

### Describe the solution you'd like

Basically I would like the reverse of `delete_matching_keys` i.e. `keep_matching_keys` which would act similar to `keep_keys`

### Describe alternatives you've considered

I ended up just doing a `set(span.attributes["myEvent"], attributes) where name == "myEvent"` in the meantime.

**Comment by TylerHelmuth** [comment]
Feels reasonable, we see this issue come up a lot where users need a subset of a map but don't know the exact keys.  Add since Go regex doesn't support negative lookahead I don't believe that `delete_matching_keys` can be used instead.

**Comment by bacherfl** [comment]
I would like to work on this one if this is still available. CC @evan-bradley
